src/main/kotlin/org/rust/lang/core/resolve/KnownItems.kt

```diff
@@ -125,3 +125,31 @@ private class RealKnownItemsLookup(
         }.orElse(null)
     }
 }
+
+enum class KnownDerivableTrait(
+    private val resolver: (KnownItems) -> RsTraitItem?,
+    val dependencies: Array<KnownDerivableTrait> = emptyArray()
+) {
+    Clone({ it.Clone }),
+    Copy({ it.Copy }, arrayOf(Clone)),
+    Debug({ it.Debug }),
+    Default({ it.Default }),
+    Hash({ it.Hash }),
+    PartialEq({ it.PartialEq }),
+    Eq({ it.Eq }, arrayOf(PartialEq)),
+    PartialOrd({ it.PartialOrd }, arrayOf(PartialEq)),
+    Ord({ it.Ord }, arrayOf(PartialOrd, Eq, PartialEq)),
+
+    Serialize({ it.findTrait("serde::Serialize") }),
+    Deserialize({ it.findTrait("serde::Deserialize") }),
+
+    // TODO Fail also derives `Display`. Ignore it for now
+    Fail({ it.findTrait("failure::Fail") }),
```

Comment:
Looks like doesn't work with `Fail`

Reply:
Hmm, but I tested it =/

Reply:
Works for me
![peek 2019-01-22 19-17](https://user-images.githubusercontent.com/3221931/51548923-5a386700-1e7a-11e9-93ec-8387b03c810f.gif)

Reply:
Resolve works as expected, but completion doesn't work for `Fail`.
It's interesting because everything is ok for `Serialize`/`Deserialize` traits

Reply:
FIxed